app/controllers/activities_controller.rb

```diff
@@ -30,4 +30,28 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def filtering_params
+    params.permit(:user_id, :trackable_type, :since, :before)
+  end
+
+  def filter_activities(activities)
+    filtering_params.each do |key, value|
+      next if value.blank?
+
+      if key == 'since'
+        value = DateTime.parse(value).beginning_of_day rescue ArgumentError
```

Comment:
Using inline `rescue` looks clean but I'm not sure if it's ideal. Can we check the valid date with `Date.valid_date?` instead?